tgui/packages/tgui/interfaces/PowerMonitor/type.ts

```diff
@@ -67,3 +68,45 @@ export interface PowerMonitorSmesStaticItemData {
 export const isDataForApc = (data: PowerMonitorData): data is PowerMonitorApcData => data.type === PowerMonitorType.Apc;
 export const isDataForSmes = (data: PowerMonitorData): data is PowerMonitorSmesData =>
   data.type === PowerMonitorType.Smes;
+
+export enum ApcTableHeaderColumns {
+  Area = 0,
+  Equipment = 1,
+  Lighting = 2,
+  Environment = 3,
+  Load = 4,
+  CellCharge = 5,
+  CellState = 6
+}
+
+export type ApcTableHeaderColumnSortState = {
+  dir: SortDirection,
+  field: ApcTableHeaderColumns
+}
+
+export enum SmesTableHeaderColumns {
+  Area = 0,
+  StoredPower = 1,
+  Charging = 2,
+  Input = 3,
+  Output = 4,
+  Active = 5,
+  Load = 6
+}
+
+export type SmesTableHeaderColumnSortState = {
+  dir: SortDirection,
+  field: SmesTableHeaderColumns
+}
+
+export const numericCompare = (a: number, b: number): number => {
```

Comment:
Is this really not a default JavaScript thing

Reply:
I would have hoped so and maybe there is, but I haven't found it if it exists. This could also probably be reduced to a-b and get the same results. I'm following the same approach in #18246 so this could either be reduced to (a - b) in both implementations or keep this implementation but make it common to both to avoid duplication, assuming there's no built in numeric compare-to function somewhere. I'm not the most fluid with javascript yet, I've worked mostly in C++ / .NET where for sure there are STL / standard library funcs that do this.

Reply:
If you don't provide a comparator, it'll compare in a way that makes sense for number/strings. One option is to either not pass an argument in, or to set the comparator function to `undefined` if using some kind of configuration or whatever where you're always passing something in to `.sort`.

A downside of using the `undefined` method is that it's not necessarily intuitive what it's doing, and easy to confuse with a "this column cannot be sorted" concept.

Some middle-ground would look like having a set of standard comparators somewhere and exporting named comparators from there (e.g. `numericComparator`) and having *that* set to `undefined` (with a comment).